### ISSUE TITLE
E2E tests of Chrome, Firefox, Safari, etc.

### ISSUE THREAD

**Issue by wi1dcard** [wi1dcard/fingerproxy#10]
## Notes

- Runs on GitHub Actions
- Maybe using puppeteer or selenium?
- Related to #8

**Comment by wi1dcard** [comment]
Wow, that's amazing!! Actually, I did some research of common fingerprints too. But only runs on GitHub Actions, and collected just a few programming languages + HTTP client libraries. Here it is: <https://github.com/wi1dcard/fingerprint-db>

I didn't write any doc or usage for that project, it is really just a simple PoC 😂 if you are interested in it, we can build something together.

**Comment by wi1dcard** [comment]
> I mean I prefer finding the most popular fingerprints and create libraries for mimicry purposes

Hahaha, you know what, I collected these fingerprints for mitigating bot abuse of free query APIs which are made for our web frontend. That's the initial propose of creating fingerproxy 😂 However, it's okay to use it from the other side, I don't mind that at all.

> especially the non-Linux platforms (which are the most popular)

Yeah, it might take some time to make all these cross-platform tests work. I'm familiar with macOS and Linux but not Windows.

**Comment by wi1dcard** [comment]
> Edge done, a little bit dirty but working https://github.com/hellodword/tls-fingerprinting/actions/runs/8845912592

Yeah actually I took a look of the code today, it is quite complex 😂 using windows in docker + github action is really a challenge. Awesome work!

**Comment by unixfox** [comment]
You can test a lot of browsers using https://www.browserstack.com.

There is even a free plan for open source project: https://www.browserstack.com/open-source

This is not an ad but an idea that I had to gather all the ja4, ja3, http2 fingerprint of many browsers.
